lang.test/build.gradle.kts

```diff
@@ -4,14 +4,24 @@ plugins {
   id("org.metaborg.spoofax.gradle.project")
 }
 
+val daggerVersion = "2.25.2"
+
 dependencies {
   api(platform(project(":pie.depconstraints")))
+  api("com.google.dagger:dagger:$daggerVersion")
```

Comment:
Please add the dagger dependencies to `pie.depconstraints`, and then add:
```
annotationProcessor(platform(project(":pie.depconstraints")))
testAnnotationProcessor(platform(project(":pie.depconstraints")))
``` 
here to be able to depend on dagger without indicating a version.

Reply:
moving dagger to :pie.depconstraints fails, it seems that it doesn't actually generate the dagger classes:
```
/Users/ivo/programming/devenv/pie/lang.test/src/test/java/mb/pie/lang/test/supplier/createAndGet/nested/NestedTest.java:3: error: cannot find symbol
import mb.pie.lang.test.supplier.createAndGet.typeString.DaggertypeStringComponent;
                                                        ^
  symbol:   class DaggertypeStringComponent
  location: package mb.pie.lang.test.supplier.createAndGet.typeString
```
Dependencies in lang.test/build.gradle.kts:
```
dependencies {
  api(platform(project(":pie.depconstraints")))
//  api("com.google.dagger:dagger:$daggerVersion") //old
//  api(platform(project(":pie.depconstraints"))) // already added at the first line
//  api("com.google.dagger:dagger") // fails (can't find "com.google.dagger:dagger:")
//  annotationProcessor("com.google.dagger:dagger-compiler:$daggerVersion") // old
  annotationProcessor(platform(project(":pie.depconstraints"))) // new
//  annotationProcessor("com.google.dagger:dagger-compiler") // fails
  compileLanguage(project(":pie.lang"))
  compileOnly("org.checkerframework:checker-qual-android")
//  testAnnotationProcessor("com.google.dagger:dagger-compiler:$daggerVersion") // old
  testAnnotationProcessor(platform(project(":pie.depconstraints"))) // new
//  testAnnotationProcessor("com.google.dagger:dagger-compiler") // fails
  testCompileOnly("org.checkerframework:checker-qual-android")
  testImplementation("org.metaborg:resource")
  testImplementation(project(":pie.runtime"))
  testImplementation(project(":pie.lang.runtime.java"))
  testImplementation(project(":pie.dagger"))
}
```
Dependencies in depconstraints:
```
dependencies {
  constraints {
    // Main
    api("org.metaborg:resource:0.7.1")
    api("org.checkerframework:checker-qual-android:3.0.0") // Use android version: annotation retention policy is class instead of runtime.
    // Test
    api("org.junit.jupiter:junit-jupiter-api:5.2.0")
    api("com.nhaarman.mockitokotlin2:mockito-kotlin:2.2.0")
    api("com.google.jimfs:jimfs:1.1")
    // this is new:
    val daggerVersion = "2.25.2"
    api("com.google.dagger:dagger:$daggerVersion")
    api("com.google.dagger:dagger-compiler:$daggerVersion")
  }
}
```

Reply:
You have to *add* the dependencies, not replace them. You have no `annotationProcessor` nor `testAnnotationProcessor` dependency to dagger. Something like:

```
dependencies {
  api(platform(project(":pie.depconstraints")))
  annotationProcessor(platform(project(":pie.depconstraints")))
  testAnnotationProcessor(platform(project(":pie.depconstraints")))

  api("com.google.dagger:dagger")
  annotationProcessor("com.google.dagger:dagger-compiler")
  compileLanguage(project(":pie.lang"))
  compileOnly("org.checkerframework:checker-qual-android")

  testAnnotationProcessor("com.google.dagger:dagger-compiler")
  testCompileOnly("org.checkerframework:checker-qual-android")
  testImplementation("org.metaborg:resource")
  testImplementation(project(":pie.runtime"))
  testImplementation(project(":pie.lang.runtime.java"))
  testImplementation(project(":pie.dagger"))
}

```